stdlib/sql/sql_test.flux

```diff
@@ -13,42 +14,74 @@ pgDsn = "postgresql://postgres@127.0.0.1:5432/postgres?sslmode=disable"
+// At time of writing the DDL used to create the tables has unquoted column names, so
+// for engines such as Snowflake and SAP HANA (hdb) we need to UPPERCASE these identifiers
+// here so the quoted version matches the table definition.
+stanley = {name: "Stanley", age: 15, "fav food": "chicken"}
+STANLEY = {NAME: "Stanley", AGE: 15, "FAV FOOD": "chicken"}
 
-// Some db engines will UPPERCASE table/column names when the identifiers are unquoted
-STANLEY = {NAME: "Stanley", AGE: 15}
+lucy = {name: "Lucy", age: 14}
 LUCY = {NAME: "Lucy", AGE: 14}
-SOPHIE = {NAME: "Sophie", AGE: 15}
+
+sophie = {name: "Sophie", age: 15, "fav food": "salmon"}
+SOPHIE = {NAME: "Sophie", AGE: 15, "FAV FOOD": "salmon"}
+
+nonseed_want = array.from(rows: [sophie])
+NONSEED_WANT = array.from(rows: [SOPHIE])
+
+// SQL Injection attempt simulation.
+// Try to write the row (each crafted for a particular dialect) to a new table.
+// Flux will try to create automatically and in the process, drop the seeded
+// table, or not if the injection is mitigated.
+// If the injection is successful, the "read_from_seed" tests should fail.
+evil = array.from(rows: [{"x\" INT);drop table \"pet info\";--": 1}])
+EVIL = array.from(rows: [{"x\" INT);drop table \"PET INFO\";--": 1}])
+myevil = array.from(rows: [{"x` INT);drop table `pet info`;--": 1}])
 
 testcase integration_hdb_read_from_seed {
-    want = array.from(rows: [STANLEY, LUCY])
-    got = sql.from(driverName: "hdb", dataSourceName: hdbDsn, query: "SELECT name, age FROM pets WHERE seeded = true")
+    got =
+        sql.from(
+            driverName: "hdb",
+            dataSourceName: hdbDsn,
+            // n.b. we must explicitly UPPER CASE the table name here to match the DDL.
+            query: "SELECT name, age, \"FAV FOOD\" FROM \"PET INFO\" WHERE seeded = true",
+        )
 
-    testing.diff(got: got, want: want)
+    testing.diff(
+        got: got,
+        want:
+            union(tables: [array.from(rows: [STANLEY]) |> debug.opaque(), array.from(rows: [LUCY]) |> debug.opaque()])
+                |> sort(columns: ["AGE"], desc: true),
+    )
         |> yield()
 }
 
 testcase integration_hdb_read_from_nonseed {
-    want = array.from(rows: [SOPHIE])
-    got = sql.from(driverName: "hdb", dataSourceName: hdbDsn, query: "SELECT name, age FROM pets WHERE seeded = false")
+    got =
+        sql.from(
+            driverName: "hdb",
+            dataSourceName: hdbDsn,
+            // n.b. we must explicitly UPPER CASE the table name here to match the DDL.
+            query: "SELECT name, age, \"FAV FOOD\" FROM \"PET INFO\" WHERE seeded = false",
+        )
+
+    testing.diff(got: got, want: NONSEED_WANT)
+        |> yield()
+}
 
-    testing.diff(got: got, want: want)
+testcase integration_hdb_injection {
+    EVIL
+        |> sql.to(driverName: "hdb", dataSourceName: hdbDsn, table: "injection attempt", batchSize: 1)
+        |> filter(fn: (r) => false)
         |> yield()
 }
```

Comment:
Is this a test case with no `diff`? How would this fail?

Reply:
I guess you just want to show that the test case succeeds with no errors, even if you don't verify the data write to the database?

Reply:
This is something that makes these tests fragile imo. I tried to explain these with a block comment here:

https://github.com/influxdata/flux/pull/4328/files#diff-9a8520c5dc40773f27979e736664f3082b8257ba14b495f2c616319eb8ed6884R33-R37

Since _writing_ to a database with `sql.to` is not something we can verify the outcome of within a single test, we rely on others after the fact to bare witness to this. Subsequent tests will fail if this test case succeeds but our injection mitigation doesn't hold.

The test _can fail_ if the client code in `sql.to` produces an error, such as not being able to connect to the database, or not having sufficient `GRANT`s to create the table if needed. For our purpose, yes. We just want to know _something was sent without error_.